registrations/serializers.py

```diff
@@ -39,6 +39,11 @@ class Meta:
                   'source', 'created_at', 'updated_at', 'created_by',
                   'updated_by')
 
+    def validate(self, attrs):
+        instance = Registration(**attrs)
+        instance.clean()
```

Comment:
I am not sure if DRF's ModelSerializer will directly call full_clean on the model instance but if not we should call it here instead of clean.